docs/src/components/ThemeExplorer/DataBlock/DataBlock.component.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+    Renders the value given to it with modification depending on the data type and content.
+
+    Strings containing hex or rgba values are rendered with a small color box next to them.
+*/
+
+import * as React from 'react';
+import styled, { css } from '@xstyled/styled-components';
+
+const colorRegex = /#[a-zA-Z0-9]{4,7}|[rgba|rgb]*\(.*\)/;
+
+const extractColor = (val: string | number | object) => {
+    if (typeof val === 'string') {
+        const check = colorRegex.exec(val) || false;
+
+        if (check) {
+            return check[0];
+        }
+    }
+    return false;
+};
+
+const ColorValue = styled('span')`
+    display: inline-block;
+    vertical-align: middle;
+`;
+
+interface ColorBoxProps {
+    backgroundColor: string;
+}
+
+const ColorBox = styled('span')<ColorBoxProps>`
+    width: 0.625rem;
+    height: 0.625rem;
+    display: inline-block;
+    vertical-align: middle;
+    border: solid black 1px;
+    margin-left: 0.25rem;
+    ${({ backgroundColor }) => css({ backgroundColor })}
+`;
+
+interface DataBlockProps {
+    data: string | number | object;
+}
+
+export const DataBlock = ({ data }: DataBlockProps) => {
+    const hexColor = React.useMemo(
+        () => (typeof data === 'string' ? extractColor(data) : false),
+        [data]
+    );
+
+    return (
+        <span>
+            {typeof data === 'function' ? (
+                // This is a little hacky. This is most likely true, that css() is being used if the
+                // type is function. It is truthy as of right now (v1.3.7)
+                <a
+                    href="https://xstyled.dev/docs/migrate-from-styled-components/#css-utility-is-required"
+                    target="_blank"
+                >
+                    css()
```

Comment:
We had talked about what to do here and agreed that perhaps it was enough to let the user know `css()` was being used here, though not technically correct. I can put that it's a function instead to future-proof. I'm concerned that it'll lead to trying to explain to users what's going on with `Button`.

Reply:
Bah, stupid ctrl+enter commented rather than going to a new line. Stupid different RTE's. :)

I will take a stab at showing `func()` instead with a brief explanation as to how `Button` uses it.